frontend/src/scenes/insights/utils/api.ts

```diff
@@ -0,0 +1,26 @@
+import api from 'lib/api'
+
+import { getInsightFilterOrQueryForPersistance } from '~/queries/nodes/InsightQuery/utils/queryNodeToFilter'
+import { getQueryBasedInsightModel } from '~/queries/nodes/InsightViz/utils'
+import { InsightModel, QueryBasedInsightModel } from '~/types'
+
+type ReturnedInsightModelByFlag<Flag extends boolean> = Flag extends true ? QueryBasedInsightModel : InsightModel
+
+export const insightsApi = {
```

Comment:
I initially did the handling in `lib/api.ts`, but that doesn't work as (1) it's included in the toolbar and importing anything there explodes size (2) doing so breaks kea, likely due to some logics needing to be mounted in the right order. Therefore this wrapper.